src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -98,35 +124,138 @@ public CourseScoresDTO calculateCourseScores(long courseId, Collection<Long> stu
+
+            List<PlagiarismCase> plagiarismCasesOfExerciseType = plagiarismCases.stream().filter(plagiarismCase -> plagiarismCase.getExercise().getExerciseType() == exerciseType)
+                    .toList();
+
+            StudentScoresDTO studentScoresOfExerciseType = calculateCourseScoreForStudent(course, userId, studentParticipationsOfExerciseType,
+                    maxAndReachablePointsOfExerciseType.maxPoints, maxAndReachablePoints.reachablePoints, plagiarismCasesOfExerciseType);
+            CourseScoresDTO scoresOfExerciseType = new CourseScoresDTO(maxAndReachablePointsOfExerciseType.maxPoints, maxAndReachablePointsOfExerciseType.reachablePoints,
+                    studentScoresOfExerciseType);
+            scoresPerExerciseType.put(exerciseType.getExerciseTypeAsString(), scoresOfExerciseType);
+        }
+
+        return scoresPerExerciseType;
     }
 
     /**
      * Calculates the presentation score, relative and absolute points for the given studentId and corresponding participationsOfStudent
      * and takes the effects of related plagiarism verdicts on the grade into account.
      *
-     * @param studentId                  the id of the student who has participated in the course exercises
-     * @param participationsOfStudent    should be non-empty. The exercise participations of the given student
-     * @param maxPointsInCourse          max points in the given course
-     * @param reachableMaxPointsInCourse max points achievable in the given course depending on the due dates of the exercises
-     * @param plagiarismMapping          the plagiarism verdicts for the student for the participated exercises
-     * @return a StudentScore instance with the presentation score, relative and absolute points achieved by the given student and the most severe plagiarism verdict
+     * @param course                     the course the scores are calculated for.
+     * @param studentId                  the id of the student who has participated in the course exercises.
+     * @param participationsOfStudent    should be non-empty. The exercise participations of the given student.
+     * @param maxPointsInCourse          max points in the given course.
+     * @param reachableMaxPointsInCourse max points achievable in the given course depending on the due dates of the exercises.
+     * @param plagiarismCases            the plagiarism verdicts for the student.
+     * @return a StudentScoresDTO instance with the presentation score, relative and absolute points achieved by the given student.
      */
-    public CourseScoresDTO.StudentScore calculateCourseScoreForStudent(Long studentId, List<StudentParticipation> participationsOfStudent, double maxPointsInCourse,
-            double reachableMaxPointsInCourse, PlagiarismMapping plagiarismMapping) {
+    private StudentScoresDTO calculateCourseScoreForStudent(Course course, Long studentId, List<StudentParticipation> participationsOfStudent, double maxPointsInCourse,
```

Comment:
I'd advise exploring if using the `ParticipantScore` to get the latest rated result speeds this method up. It should be faster than fetching the Participation and iterating over all Results to find the latest.